### ISSUE TITLE
IDISP002 & IDISP006 should not warn here.

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#150]
```cs
public class MarkdownFile
{
    public MarkdownFile(string name)
    {
        this.Name = name;
        if (File.Exists(name))
        {
            this.AllText = File.ReadAllText(name);
            this.AllLines = File.ReadAllLines(name);
        }
    }

    public string Name { get; }

    public bool Exists => File.Exists(this.Name);

    public string AllText { get; }

    public IReadOnlyList<string> AllLines { get; }
}
```
